pf/proto/attribute.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proto
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+)
+
+var _ = shim.Schema(attribute{})
+
+type attribute struct{ attr tfprotov6.SchemaAttribute }
+
+// Simple schema options
+
+func (a attribute) Optional() bool      { return a.attr.Optional }
+func (a attribute) Required() bool      { return a.attr.Required }
+func (a attribute) Description() string { return a.attr.Description }
+func (a attribute) Computed() bool      { return a.attr.Computed }
+func (a attribute) ForceNew() bool      { return false }
```

Comment:
Deserve a comment why it's ok to return false?

Reply:
I'm happy to add one. It's because this information isn't exposed by terraform at the gRPC level.

Reply:
Trying to remember if we use it for docs-only or use it to drive behavior?

Reply:
This is the only place where it impacts behavior:

https://github.com/pulumi/pulumi-terraform-bridge/blob/6d24ffc501f1b9757b85548b92f5e837e2e30994/pkg/tfbridge/provider.go#L1053-L1063

I think that's just used for display logic.

Reply:
I think stables something deprecate-able. So all good then.